nvflare/apis/impl/controller.py

```diff
@@ -287,7 +287,7 @@ def process_task_request(self, client: Client, fl_ctx: FLContext) -> Tuple[str,
                 task.last_client_task_map[client.name] = client_task_to_send
                 task.client_tasks.append(client_task_to_send)
                 self._client_task_map[client_task_to_send.id] = client_task_to_send
-            return task_name, client_task_to_send.id, task_data
+            return task_name, client_task_to_send.id, copy.deepcopy(task_data)
```

Comment:
This may be using lots of memory as task_data is a shareable.

Reply:
I am aware of this. Ideally, we need only the copy the headers part of the shareable. But we don't that enabled in the Sharable. We can enhance that in the future.

Reply:
The upstream (who called this method) is trying to set header to this task_data's shareable.
Resulting in multiple threads trying to write/set same key to different values at the same time.

By returning a full copy, we don't write/set the same object.